db/sql/sql.go

```diff
@@ -870,7 +870,12 @@ func (tx *Transaction) executeSelect(ctx context.Context, sc *selectContext, sql
 	if err != nil {
 		return nil, 0, err
 	}
-	total, err = tx.Count(ctx, sc.schema, sc.filter)
+
+	if sc.paginator != nil && (sc.paginator.Offset != 0 || sc.paginator.Limit != 0) {
```

Comment:
Why `sc.paginator.Limit != 0` and not `sc.paginator.Limit != math.MaxUint64`? Default value for `paginator.Limit` is `math.MaxUint64`

Please make test with
`pagination.NewPaginator(pagination.OptionLimit(0))`

Reply:
how about
```go
func (sc selectContext) hasPagination() bool
```
?

Reply:
No idea how I've missed that, THANK YOU!

Added a test for the "LIMIT 0" case.

Reply:
@p-kozlowski went with `func (tx *Transaction) isSelectPaginated`, because it's used only in `func (tx *Transaction) executeSelect` and `selectContext` has no methods at all, it's always used as a parameter.